brave/brave-core
# Use Rust HTTPS Everywhere library

Resolves https://github.com/brave/brave-browser/issues/5280.

Blocked on acceptance of https://github.com/brave/https-everywhere-builder/pull/40 to ship the new format accepted by the rust library.

I've generated a private key, public key, and component ID for the new v7 component according to the instructions on https://github.com/brave/brave-core-crx-packager#development, however this is not currently uploaded anywhere because I don't have permissions to do so. If it's preferred for someone else to generate/upload these keys I am happy to update this PR.

Users who are upgrading from previous versions of Brave will automatically uninstall the v6 component and upgrade to the new v7 component; users who are installing for the first time or have already migrated will just load the v7 component normally.

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

The following steps should be tested:
 - Without any existing profile directory
 - With an existing profile from a previous version of Brave
 - With an existing profile from this version of Brave (i.e. one of the above after a browser restart, just be sure to re-enable HTTPS Everywhere in the Shields panel)

1. On Desktop, navigate to http://https-everywhere.badssl.com.
2. Ensure the page appears with a checkmark at the bottom that says "HTTPS Everywhere is working".
3. Ensure that the Shields icon shows 1 on the counter. Opening the Shields panel in advanced mode should show 1 connection upgraded to HTTPS. Clicking the dropdown on the 1 should show the single entry, `https-everywhere.badssl.com`.
4. Disabling the "Connections upgraded to HTTPS" toggle in Shields should result in the page refreshing. After the reload, the checkmark should be replaced by a red X, labeled "HTTPS Everywhere is not working".
5. Close the browser and verify that the profile directory does _not_ contain a directory named `oofiananboodjbbmdelgdommihjbkfag`.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
